osidb/models/affect.py

```diff
@@ -511,6 +525,21 @@ def _validate_purl_and_ps_component(self, **kwargs):
                     f"ps_component: {self.ps_component}, purl: {self.purl}."
                 )
 
+    def _validate_not_affected_justification(self, **kwargs):
+        """
+        The not affected justification field becomes mandatory if the affectedness is NOTAFFECTED
+        and there are no trackers to compute the justification from.
+        If the affectedness is set at a different value than NOTAFFECTED, then the justification should
+        be empty.
+        """
+        if (
+            self.affectedness == Affect.AffectAffectedness.NOTAFFECTED
+            and not self.not_affected_justification
+        ):
+            raise ValidationError(
+                f"Affect ({self.uuid}) is set as NOTAFFECTED but the 'not affected justification' is empty."
+            )
```

Comment:
I believe that you have a plan for it but just mentioning that the eventual release of this must be coordinated with (at least) OSIM or otherwise nobody will be able to set affects as NOTAFFECTED.

Reply:
Yes I'm aware, this should not be released until the other parts have been fully implemented